sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -1410,6 +1410,21 @@ class AnalysisSuite extends AnalysisTest with Matchers {
     assertAnalysisSuccess(finalPlan)
   }
 
+  test("Remove extra project as long as all attributes are semantically the same") {
+    val analyzer = getAnalyzer
+    val inner = Project(
+      Seq(
+        Alias(testRelation2.output(0), "a")(),
+        testRelation2.output(1),
+        Alias(testRelation2.output(2), "c")(),
+        testRelation2.output(3),
+        testRelation2.output(4),
+      ),
+      testRelation2)
+    val actual_plan = analyzer.simplifyPlanForCollectedMetrics(inner)
```

Comment:
note: now we only call `simplifyPlanForCollectedMetrics` with canonicalized plan. We should match it in the tests.